images/win/scripts/Installers/Install-Msys2.ps1

```diff
@@ -53,40 +53,38 @@ pacman.exe -Syuu  --noconfirm
 taskkill /f /fi "MODULES eq msys-2.0.dll"
 
 Write-Host "`n$dash Install msys2 packages"
-pacman.exe -S --noconfirm --needed --noprogressbar base-devel compression
+$msysPackages = (Get-ToolsetContent).MsysPackages.msys2
+pacman.exe -S --noconfirm --needed --noprogressbar $msysPackages
 taskkill /f /fi "MODULES eq msys-2.0.dll"
 
 Write-Host "`n$dash Remove p7zip/7z package due to conflicts"
 pacman.exe -R --noconfirm --noprogressbar p7zip
 
-# mingw package list
-# libxml2 can be removed from the list after the issue is fixed https://github.com/msys2/MINGW-packages/issues/8658
-$tools64 = "___clang ___clang-tools-extra ___cmake ___libxml2 ___llvm ___toolchain ___ragel"
-$tools32 = "___clang ___cmake ___libxml2 ___llvm ___toolchain ___ragel"
-
-# install mingw64 packages
-Write-Host "`n$dash Install mingw64 packages"
-$pre = "mingw-w64-x86_64-"
-pacman.exe -S --noconfirm --needed --noprogressbar $tools64.replace('___', $pre).split(' ')
-
-# install mingw32 packages
-Write-Host "`n$dash Install mingw32 packages"
-$pre = "mingw-w64-i686-"
-pacman.exe -S --noconfirm --needed --noprogressbar $tools32.replace('___', $pre).split(' ')
+# install mingw packages
+$archs = (Get-ToolsetContent).MsysPackages.mingw.arch
+foreach ($arch in $archs)
+{
+  Write-Host "Installing $arch packages"
+  $runtimePackages = ((Get-ToolsetContent).MsysPackages.mingw | Where-Object { $_.arch -eq $arch }).runtime_packages.name | ForEach-Object { "${arch}-$_" }
```

Comment:
nitpick: use a new variable for `(Get-ToolsetContent).MsysPackages.mingw | Where-Object { $_.arch -eq $arch }`

Reply:
Nice, thanks!